### PR TITLE
Convert string input on calculate qc metrics into collection for convenience

### LINKED ISSUE #2858: String conversion of `qc_vars` into a `Collection` when a `str` input is provided in `calculate_qc_metrics`
### What kind of feature would you like to request?

Additional function parameters / changed functionality / changed defaults?

### Please describe your wishes

I often make the mistake of providing a single string input to the `qc_vars` argument in `calculate_qc_metrics` which always fails with a `KeyError` because it beings to iterate over the string. This could be a very simple addition to just convert the argument into a `list` if a `str` is provided as input.

### PR DESCRIPTION
**Pull request by noamteyssier**
- [x] Closes #2858 
- [x] Tests included or not required because:

- [x] Release notes not necessary because: this is a minor quality of life change, and could be tacked onto any other release as opposed to creating a dedicated release

### CODE REVIEW

File: scanpy/tests/test_qc_metrics.py [hunk 1]
```diff
@@ -143,6 +143,19 @@ def test_qc_metrics_format(cls):
         assert np.allclose(adata.var[col], adata_dense.var[col])
 
 
+@pytest.mark.parametrize(
+    "cls", [np.asarray, sparse.csr_matrix, sparse.csc_matrix, sparse.coo_matrix]
+)
```
Comment (ivirshup) [inline scanpy/tests/test_qc_metrics.py]:
```suggestion
```

I think it's simple enough that we don't need to parametrize this test

**Review by ivirshup — COMMENTED** [review]
LGTM, thanks for the contribution.

Could you add a release note?

**Comment by noamteyssier** [comment]
Sounds good - added an entry to the release note and updated the tests to not use the parameterization.

**Review by ivirshup — APPROVED** [review]

**Comment by ivirshup** [comment]
Thanks! I've moved the release note to 1.10, since this is more of a feature.

**Pull request merged**